### ISSUE TITLE
输出结果一直重复，只到max_new_tokens长度

### ISSUE THREAD

**Issue by liuliuNLP** [QwenLM/Qwen2#779]
### model
model_name_or_path: model/Qwen2-1.5B

### method
stage: sft
do_train: true
finetuning_type: full

### ddp
ddp_timeout: 180000000
deepspeed: examples/deepspeed/ds_z0_config.json

### dataset
dataset: ner_ee_ie
template: qwen
cutoff_len: 2048
max_samples: 2000000
overwrite_cache: true
preprocessing_num_workers: 16

### output
output_dir: saves/qwen2-0.5b/full/sft
logging_steps: 10
save_steps: 500
plot_loss: true
overwrite_output_dir: true

### train
per_device_train_batch_size: 1
gradient_accumulation_steps: 2
learning_rate: 1.0e-4
num_train_epochs: 3.0
lr_scheduler_type: cosine
warmup_ratio: 0.01
fp16: true

### eval
val_size: 0.1
per_device_eval_batch_size: 1
eval_strategy: steps
eval_steps: 500

全量微调是不是过拟合了，推理结果如下：
<img width="748" alt="企业微信截图_17210966351194" src="https://github.com/user-attachments/assets/65b2fe56-d55e-4a94-975b-601896dc7dd4">
显示警告：
Special tokens have been added in the vocabulary, make sure the associated word embeddings are fine-tuned or trained.
The attention mask and the pad token id were not set. As a consequence, you may observe unexpected behavior. Please pass your input's `attention_mask` to obtain reliable results.
Setting `pad_token_id` to `eos_token_id`:151643 for open-end generation.
The attention mask is not set and cannot be inferred from input because pad token is same as eos token.As a consequence, you may observe unexpected behavior. Please pass your input's `atteion_mask` to obtain reliable results.

**Comment by jklj077** [comment]
if you finetune the base model, try changing the eos token/eos token id in config.json and the generation_config.json to <|im_end|>/151645

**Comment by liuliuNLP** [comment]
<img width="769" alt="企业微信截图_17210984051808" src="https://github.com/user-attachments/assets/33ba5b3e-d02a-481b-b2cd-92137e274d38">
还是这样，他提示这个Special tokens have been added in the vocabulary, make sure the associated word embeddings are fine-tuned or trained.

**Comment by jklj077** [comment]
what was the library you were using?

**Comment by liuliuNLP** [comment]
可以了，谢谢

**Comment by shilei4260** [comment]
@liuliuNLP 如果您对基础模型进行了配置，请尝试将 config.json 中的 eos token/eos token id 和 Generation_config.json 更改为 <|im_end|>/15164 <|im_end|>加入到哪里呀，我用的是Qwen2-7B 不是Qwen2-7B-Instruct  因为Qwen2-7B 是128k ,Qwen2-7B-Instruct 32k 用vllm推理Qwen2-7B默认的参数会一直重复直到max_new_tokens的长度，我修改tokenizer_config.json中eos_token为<|im_end|>后有时候还是会出现一直重复直到max_new_tokens的长度。
![捕获2](https://github.com/user-attachments/assets/ed2075a2-bdf1-4b16-98ad-4b304ce22aae)
![捕获3](https://github.com/user-attachments/assets/6128a271-159f-4958-aa22-f6cb9c2d02dc)

**Comment by 297106271** [comment]
怎么解决的呢？

**Comment by rookie0607** [comment]
> 可以了，谢谢

请问如何解决的 @liuliuNLP

**Comment by JoyDuomo** [comment]
> 可以了，谢谢
遇到同样问题，请问是怎么解决的呢？

**Comment by JoyDuomo** [comment]
> 可以了，谢谢

遇到同样的问题，请问是怎么解决的呢 @liuliuNLP

**Comment by wangkz2023** [comment]
> what was the library you were using? it is likely data preprocessing was not done correctly. in addition, is it 0.5B or 1.5B.

谢谢大佬
